crates/bevy_ecs/src/query/iter.rs

```diff
@@ -122,6 +122,66 @@ impl<'w, 's, D: QueryData, F: QueryFilter> QueryIter<'w, 's, D, F> {
         }
     }
 
+    /// Executes the equivalent of [`Iterator::fold`] over a contiguous segment
+    /// from an storage.
+    ///
+    ///  - `range` must be in `[0, storage::entity_count)` or None.
```

Comment:
This should probably be formatted like public facing safety docs with `/// # Safety` above it. As an aside for a later time we should consider enabling `clippy::check-private-items` to catch these.